### PR TITLE
Hex grid layer methods

### PR DESCRIPTION
**Pull request by heshan0131**
For easier subclassing:
- Add static methods to layer class
- Add method `getSubLayer ` to get SubLayer class
- Add method `getSubLayerProps` to get SubLayer props

### CODE REVIEW

**Comment by shaojingli** [comment]
@heshan0131 can you give a brief explanation why we need this change and what it resolves? is it for people to extend Hexagon / Cell layer more easily?

**Comment by heshan0131** [comment]
Yes.  It would be a lot easier to break layer class into smaller methods for Subclassing
For example, I am working on a Subclass of hexagon layer where I want to render a customized cell layer that uses customized shader. To do this, I only need to overide the getSubLayer methods, instead of rewrite the entire renderSubLayer call.

File: src/layers/core/hexagon-layer/hexagon-layer.js [hunk 1]
```diff
@@ -238,7 +239,19 @@ export default class HexagonLayer extends CompositeLayer {
...
-    });
+    };
+  }
+
+  getSubLayer() {
+    return HexagonCellLayer;
+  }
+
+  renderLayers() {
+    const SubLayerComp = this.getSubLayer();
```
Comment (shaojingli) [inline src/layers/core/hexagon-layer/hexagon-layer.js]: Can we change the name of this variable? `SubLayerComp` doesn't provide much info....

File: src/layers/core/hexagon-layer/hexagon-layer.js [hunk 2]
```diff
@@ -238,7 +239,19 @@ export default class HexagonLayer extends CompositeLayer {
       getColor: this._onGetSublayerColor.bind(this),
       getElevation: this._onGetSublayerElevation.bind(this),
       updateTriggers: this.getUpdateTriggers()
-    });
+    };
+  }
+
+  getSubLayer() {
```
Comment (shaojingli) [inline src/layers/core/hexagon-layer/hexagon-layer.js]: Also this, can we change it to `getSubLayerClass`

File: src/layers/core/hexagon-layer/hexagon-layer.js [hunk 3]
```diff
@@ -217,7 +218,7 @@ export default class HexagonLayer extends CompositeLayer {
     // viewport props
     const {positionOrigin, projectionMode, modelMatrix} = this.props;
 
-    return new HexagonCellLayer({
+    return {
```
Comment (shaojingli) [inline src/layers/core/hexagon-layer/hexagon-layer.js]: Add a comment saying `// return a bunch of props to the sublayer constructor specified in  getSubLayer()`

**Comment by heshan0131** [comment]
rebased on updated

**Pull request merged**